runtime/compiler/ilgen/Walker.cpp

```diff
@@ -3578,13 +3578,53 @@ TR_J9ByteCodeIlGenerator::genIfTwoOperand(TR::ILOpCodes nodeop)
 int32_t
 TR_J9ByteCodeIlGenerator::genIfImpl(TR::ILOpCodes nodeop)
    {
-   _methodSymbol->setHasBranches(true);
    int32_t branchBC = _bcIndex + next2BytesSigned();
    int32_t fallThruBC = _bcIndex + 3;
 
    TR::Node * second = pop();
    TR::Node * first = pop();
 
+   TR::DataType type = first->getDataType();
+   if (branchBC > _bcIndex &&
+       first->getOpCode().isLoadConst() &&
+       second->getOpCode().isLoadConst() &&
+       type != TR::Address &&
+       type != TR::Float &&
+       type != TR::Double)
+      {
+      int64_t v1 = first->getConstValue();
```

Comment:
is the widening of narrow types in this way guaranteed to always do the right thing? for equal/not equal the extension mode doesn't really matter, but for relative operations it does - do we need to respect the original type and widen 'the right way' for whatever that means?

Reply:
An integral constant, regardless of its type, is stored in `_unionBase._constValue` which is a `int64_t`.  Any conversion has been done when storing the constant to `_unionBase._constValue`. But there will be problem in unsigned comparison that I'll fix.